### PR TITLE
PAR-1743: Add document download component

### PR DESCRIPTION
**Pull request by NappiLe**
download

### CODE REVIEW

File: src/Nordea/Components/FileDownload.elm [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    | Loading
+    | Success
+
+
+init : Translate -> msg -> String -> String -> FileDownload msg
+init translate onDownload label id =
+    FileDownload
+        { onDownload = onDownload
+        , label = label
+        , type_ = Nothing
```
Comment (fredrikloberg) [inline src/Nordea/Components/FileDownload.elm]: Is this better?
```suggestion
        , subtitle = Nothing
```

File: src/Nordea/Components/FileDownload.elm [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                    [ lineHeight (rem 1.5)
+                    , Themes.color Themes.PrimaryColor Colors.deepBlue
+                    , textDecoration underline
+                    , marginLeft auto
+                    , children [ everything [ verticalAlign middle ] ]
+                    ]
+                    :: attrs
+                )
+                [ Html.text (config.translate strings.download), downloadIconStatus ]
+        ]
```
Comment (fredrikloberg) [inline src/Nordea/Components/FileDownload.elm]: I think the attrs should be on the outermost element
```suggestion
    Html.a
        ([ onClick config.onDownload
        , css
            [ displayFlex
            , alignItems center
            , borderRadius (rem 0.5)
            , border3 (rem 0.0625) solid Colors.mediumGray
            , padding (rem 1)
            , pseudoClass "not(:last-child)" [ marginBottom (rem 0.5) |> Css.important ]
            , flexBasis (pct 100)
            , cursor pointer
            ]
        ] ++ attrs)
        [ Layout.column []
            [ Text.textLight
                |> Text.view [ css [ marginRight (rem 0.5), lineHeight (rem 1.5) ] ] [ Html.text config.label ]
            , Text.textTinyLight
                |> Text.view [ css [ marginRight (rem 0.5), lineHeight (rem 1.5) ] ] [ Html.text (config.type_ |> Maybe.withDefault "") ]
            ]
        , Text.textSmallLight
            |> Text.view
                [ css
                    [ lineHeight (rem 1.5)
                    , Themes.color Themes.PrimaryColor Colors.deepBlue
                    , textDecoration underline
                    , marginLeft auto
                    , children [ everything [ verticalAlign middle ] ]
                    ]
                ]
                [ Html.text (config.translate strings.download), downloadIconStatus ]
        ]
```

File: src/Nordea/Components/FileDownload.elm [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+type FileDownload msg
+    = FileDownload (Config msg)
+
+
+type alias Config msg =
+    { onDownload : msg
+    , label : String
+    , type_ : Maybe String
+    , id : String
```
Comment (fredrikloberg) [inline src/Nordea/Components/FileDownload.elm:59]: Is this used?
```suggestion
```

File: src/Nordea/Components/FileDownload.elm [hunk 4]
```diff
@@ -0,0 +1,143 @@
+module Nordea.Components.FileDownload exposing (..)
```
Comment (fredrikloberg) [inline src/Nordea/Components/FileDownload.elm:1]: I think it is a good idea to only explicitly expose what we need and not everything

**Review by fredrikloberg — APPROVED** [review]
Looks good, just a few minor comments 🙌

File: src/Nordea/Components/FileDownload.elm (hunk 3 above)
Comment (NappiLe) [reply src/Nordea/Components/FileDownload.elm]: It doesnt use anywhere. I will remove it 😊

File: src/Nordea/Components/FileDownload.elm (hunk 4 above)
Comment (NappiLe) [reply src/Nordea/Components/FileDownload.elm]: I agree. I will change it

**Pull request merged**